core/src/main/java/org/apache/iceberg/avro/Avro.java

```diff
@@ -446,6 +447,11 @@ public ReadBuilder classLoader(ClassLoader classLoader) {
       return this;
     }
 
+    public ReadBuilder setFileSchema(Schema fileSchema) {
```

Comment:
This is only called in the tests. I am not sure how it is called in the non-test path.

Reply:
Yes, this code together with the code change in `ProjectionDatumReader.setSchema` are added only for test the case that `fileSchema` to read Avro file is different from the actual schema to write the Avro file. This test case proves that the data cannot be returned by mistake if the schema to read the file is different from the schema to write the file.  In normal code flow, the `fileSchema` to read Avro file is always read from the metadata of the actual Avro file, therefore the schema to read file is always the same as the schema to write the file.  This code change is for test only and does not affect the normal code flow. It can be removed.